cluster-autoscaler/cloudprovider/gce/gce_manager.go

```diff
@@ -634,21 +695,68 @@ type migAutoDiscoveryConfig struct {
 	MinSize int
 	// MaxSize specifies the maximum size for all MIGs that match Re.
 	MaxSize int
+	// Optional map of labels to filter MIGs by.
+	Labels map[string]string
 }
 
+// parseMIGAutoDiscoverySpec parses individual MIG auto-discovery spec.
+//
+// It takes a spec string as input and returns a migAutoDiscoveryConfig and an error.
 func parseMIGAutoDiscoverySpec(spec string) (migAutoDiscoveryConfig, error) {
-	cfg := migAutoDiscoveryConfig{}
-
 	tokens := strings.Split(spec, ":")
 	if len(tokens) != 2 {
-		return cfg, fmt.Errorf("spec \"%s\" should be discoverer:key=value,key=value", spec)
+		return migAutoDiscoveryConfig{}, fmt.Errorf("spec \"%s\" should be mig:key=value,key=value", spec)
+	}
+	switch tokens[0] {
+	case autoDiscovererTypeLabel:
+		return parseAutoDiscoveryTypeLabelSpec(tokens[1])
+	case autoDiscovererTypeMIG:
+		return parseAutoDiscoveryTypeNamePrefixSpec(tokens[1])
+	default:
+		return migAutoDiscoveryConfig{}, fmt.Errorf("unsupported auto-discovery type specified. Supported types are 'label' and 'mig'")
+	}
+}
+
+// parseAutoDiscoveryTypeLabelSpec parses the label auto discovery specification.
+//
+// It takes a spec string as input and returns a migAutoDiscoveryConfig and an error.
+// Validates and returns an error if the spec is invalid. Examples: empty labels, key-value not separated by '=' etc.
+func parseAutoDiscoveryTypeLabelSpec(spec string) (migAutoDiscoveryConfig, error) {
+	cfg := migAutoDiscoveryConfig{}
+
+	labels := make(map[string]string)
+	for _, arg := range strings.Split(spec, ",") {
+		kv := strings.Split(arg, "=")
+		if len(kv) != 2 {
+			return cfg, fmt.Errorf("invalid label key=value pair %s; use key1=value1,key2=value2", kv)
+		}
+		labels[kv[0]] = kv[1]
 	}
-	discoverer := tokens[0]
-	if discoverer != autoDiscovererTypeMIG {
-		return cfg, fmt.Errorf("unsupported discoverer specified: %s", discoverer)
+
+	if len(labels) == 0 {
+		return cfg, fmt.Errorf("no labels specified. use key1=value1,key2=value2")
 	}
+	cfg.Labels = labels
+	cfg.Re = regexp.MustCompile(".*") // match all
 
-	for _, arg := range strings.Split(tokens[1], ",") {
+	// set default min, max size. will be overridden by instance template labels if present
+	if scaleToZeroSupported {
```

Comment:
nit: This is a const that has been set to true for past 7 years, I highly doubt we'll stop supporting scale to 0 at this point.